src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/PersistableFieldBuilder.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace System.Reflection.Emit.Experiment
+{
+    internal sealed class PersistableFieldBuilder : FieldBuilder
+    {
+        private PersistableTypeBuilder _typeBuilder;
+        private string _fieldName;
+        private FieldAttributes _attributes;
+        private Type _fieldType;
+
+        internal PersistableFieldBuilder(PersistableTypeBuilder typeBuilder, string fieldName, Type type,
+            Type[]? requiredCustomModifiers, Type[]? optionalCustomModifiers, FieldAttributes attributes)
+        {
+            ArgumentException.ThrowIfNullOrEmpty(fieldName);
+
+            ArgumentNullException.ThrowIfNull(type);
+
+            if (type == typeof(void))
```

Comment:
This is not very useful check. This will be often used with foreign type system that will have `void` represented by their own type object.

I would delete this check